microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/CpuLoadTest.java

```diff
@@ -0,0 +1,146 @@
+        String mpConfig = "hystrix.command.default.execution.timeout.enabled=true";
+        return ShrinkWrap.create(WebArchive.class, CpuLoadTest.class.getSimpleName() + ".war")
+                .addPackage(LoadService.class.getPackage())
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    @BeforeClass
+    public static void serverSetup() throws Exception {
+        MicroProfileFaultToleranceServerConfiguration.enableFaultTolerance();
+    }
+
+    /**
+     * @tpTestDetails Deploy MP FT application and call service with @Timeout annotation. During execution
+     *                of this method cause 10s CPU load.
+     * @tpPassCrit Verify that method throws {@link org.eclipse.microprofile.faulttolerance.exceptions.TimeoutException}
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testCpuLoadWithTimeout() throws Exception {
+        Assume.assumeNotNull("This test cannot be executed on this platform as ProcessUtils class was not " +
+                "implemented for it.", processUtils);
+
+        // call service which takes 5 sec
+        Future<String> underCpuLoadCall = Executors.newSingleThreadExecutor().submit(
+                () -> get(baseApplicationUrl + "?operation=timeout").body().asString());
+
+        causeCpuLoadOnServer(DEFAULT_CPU_LOAD_DURATION);
+
+        assertThat(underCpuLoadCall.get(),
+                containsString("org.eclipse.microprofile.faulttolerance.exceptions.TimeoutException"));
+    }
+
+    /**
+     * @tpTestDetails Deploy MP FT application and call service with @Retry annotation. During execution
+     *                of this method cause 10s CPU load.
+     * @tpPassCrit Verify that method throws Exception once max retry is reached.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testCpuLoadWithRetry() throws Exception {
+        Assume.assumeNotNull("This test cannot be executed on this platform as ProcessUtils class was not " +
+                "implemented for it.", processUtils);
+
+        Future<String> underCpuLoadCall = Executors.newSingleThreadExecutor().submit(
+                () -> get(baseApplicationUrl + "?operation=retry").body().asString());
+
+        causeCpuLoadOnServer(DEFAULT_CPU_LOAD_DURATION);
+
+        assertThat(underCpuLoadCall.get(), containsString("Exception from @Retry method."));
+    }
+
+    /**
+     * @tpTestDetails Deploy MP FT application and call service with @Fallback annotation. During execution
+     *                of this method cause 10s CPU load.
+     * @tpPassCrit Verify that @Fallback method was called.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void testCpuLoadWithFallback() throws Exception {
+        Assume.assumeNotNull("This test cannot be executed on this platform as ProcessUtils class was not " +
```

Comment:
Wouldn't it be better to have this assume condition in `BeforeClass`?

Reply:
+1, fixed